### PR TITLE
Backport 2.1: Add ecc extensions only if ecc ciphersuite is used

### PR DESCRIPTION
**Pull request by RonEld**
## Description
Backport of #1378 to mbedtls-2.1

## Status
**READY**

### CODE REVIEW

File: tests/ssl-opt.sh [hunk 1]
```diff
@@ -3739,106 +3739,55 @@ run_test    "Large packet TLS 1.2 AEAD shorter tag" \
             -c "16384 bytes written in 1 fragments" \
             -s "Read from client: 16384 bytes read"
 
-# Tests for DTLS HelloVerifyRequest
+# Tests for ECC extensions (rfc 4492)
```
Comment (mpg) [inline tests/ssl-opt.sh]: Shouldn't this come before the DTLS tests, rather than replacing them?

Comment (RonEld) [reply tests/ssl-opt.sh]: I think it does come before the DTLS tests. I think this is just a diff alignment issue, isn't it?

Comment (RonEld) [reply tests/ssl-opt.sh]: Yes, I believe it was a merge conflict, that these lines were accidentally deleted. I will return them, and put them after line  3792 which comments on DTLS tests

**Comment by RonEld** [comment]
@mpg @andresag01 I have restored the lines that were accidentally deleted in previous merge conflict.
please review

File: library/ssl_cli.c [hunk 2]
```diff
@@ -829,6 +833,11 @@ static int ssl_write_client_hello( mbedtls_ssl_context *ssl )
         MBEDTLS_SSL_DEBUG_MSG( 3, ( "client hello, add ciphersuite: %2d",
                        ciphersuites[i] ) );
 
+#if defined(MBEDTLS_ECDH_C) || defined(MBEDTLS_ECDSA_C) || \
+    defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED)
```
Comment (andresag01) [inline library/ssl_cli.c]: I do not think that Mbed TLS 2.1 has ECJPAKE. Does it? If it does not, are we sure we want to keep these preprocessor conditions with `defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED)`? I dont see a problem with leaving it in other than is a bit unnecessary and slightly inconsistent.

Comment (RonEld) [reply library/ssl_cli.c]: good catch!
It was inserted by mistake when I cherry=picked the changes. I thought I removed all references to ECJPAKE

**Comment by RonEld** [comment]
@andresag01 I addressed your comment. Please review.

cc @mpg

**Pull request merged**